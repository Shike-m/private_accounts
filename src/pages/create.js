import React from 'react';
import PriceForm from '../components/priceForm';

const Create = ({ match }) => {
    return (
        <div>
            <h2>This it is the create page {match.params.id}</h2>
            {/* <PriceForm /> */}
        </div>
       )
}

export default Create;