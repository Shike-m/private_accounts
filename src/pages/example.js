import React from 'react';

const Example = ({ match }) => {

    return (
        <div>
            <h2>THis is an {match.params.name} of routers.</h2>
            <h1>For videos</h1>
            <video src='youtube.com/0011001' style={{width:'300px',height:"400px",margin:'30px'}} controls='controls'></video>
        </div>
       
        
    )
}

export default Example;