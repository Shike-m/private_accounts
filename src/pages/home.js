import React, { Component, Fragment } from 'react';
import PriceList from '../components/priceList';
// import ViewTab from '../components/viewTab';
import TotalPrice from '../components/totalPrice';
import MonthPicker from '../components/monthPicker';
import CreateBtn from '../components/createBtn';
import { Tabs, Tab } from '../components/tabs';
// import Loader from '../components/loader';
import { LIST_VIEW, CHART_VIEW, TYPE_INCOME, TYPE_OUTCOME } from '../constants';
import { parseToYearAndMonth, padLeft } from '../utility';
// import AppContext from '../AppContext';
import { withContext } from '../HOC/withContext';
import { withRouter } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css'
import logo from '../logo.svg';
import '../App.css';

export const categories = {
    "1": {
        "id": 1,
        "name": "trip",
        "type": "outcome",
        "iconName": 'car'
    },
    "2": {
        "id": 2,
        "name": "investment",
        "type": "income",
        "iconName": 'coffee'
    }
}

export const items = [{
    "id": 1,
    "title": "Travel to abroad",
    "price": 100,
    "date": "2019-04-03",
    "cid": 1
}, {
    "id": 2,
    "title": "Travel to abroad",
    "price": 200,
    "date": "2019-04-03",
    "cid": 1
}
]
const newItem = {
    "id": 3,
    "title": "sales",
    "price": 200,
    "date": "2019-04-06",
    "cid": 2
}

const tabsText = [LIST_VIEW, CHART_VIEW];

class Home extends Component {

    constructor(props) {
        super(props);
        this.state = {
            items,
            currentDate: parseToYearAndMonth(),
            tabView: LIST_VIEW
        }
    }

    changeView = (view) => {
        this.setState({
            tabView: tabsText[view]
        })
    }

    changeDate = (year, month) => {
        this.setState({
            currentDate: { year, month }
        })
    }

    modifyItem = (item) => {
        this.props.history.push(`/create/${item.id}`)
        // const modifiedItems = this.state.items.map(item => {
        //     if (item.id === modifiedItem.id) {
        //         return { ...item, title: "updated" }
        //     } else {
        //         return item;
        //     }
        // });
        // this.setState({
        //     items: modifiedItems
        // })
    }

    createItem = () => {
        this.props.history.push('/create');
    }

    deleteItem = (item) => {
        this.props.actions.deleteItem(item);
        // const filterItems = this.state.items.filter(item => item.id !== deleteItem.id);
        // this.setState({
        //     items: filterItems
        // })
    }

    render() {
        const { data } = this.props;
        // console.log(data);
        const { items, currentDate, tabView } = this.state;
        // const { data } = this.props;
        // const { isLoading } = data;
        const tabIndex = tabsText.findIndex(tabText => tabText === tabView)
        const itemsWithCategory = items.map(item => {
            item.category = categories[item.cid];
            return item;
        }).filter(item => {
            return item.date.includes(`${currentDate.year}-${padLeft(currentDate.month)}`)
        })
        let totalIncome = 0, totalOutcome = 0;
        itemsWithCategory.map((item) => {
            if (item.type === TYPE_INCOME) {
                totalIncome += item.price;
            }
            if (item.type === TYPE_OUTCOME) {
                totalOutcome += item.price;
            }
        })
        return (

            <Fragment>
                <header className='app-header'>
                    <div className='row md-5 logo-position'>
                        <img src={logo} className='App-logo' alt='logo' />
                    </div>
                    <div className='row' >
                        <div className='col' >
                            <MonthPicker
                                year={currentDate.year}
                                month={currentDate.month}
                                onChange={this.changeDate}
                            />
                        </div>
                        <div className='col' style={{ height: '60px' }}>
                            <TotalPrice
                                income={totalIncome}
                                outcome={totalOutcome}
                            />
                        </div>
                    </div>
                </header>
                <div className='content-area py-3 px-3'>
                    {/* {isLoading && <Loader />} */}
                    {<React.Fragment>
                        <Tabs activeIndex={tabIndex} onTabChange={this.changeView}>
                            <Tab>List view</Tab>
                            <Tab>Chart View</Tab>
                            <Tab>test</Tab>
                        </Tabs>
                    </React.Fragment>}
                    {/* <ViewTab activeTab={tabView} onTabChange={this.changeView} /> */}
                    <CreateBtn createItem={this.createItem} />
                    {tabView === LIST_VIEW &&
                        <PriceList
                            items={itemsWithCategory}
                            onModifyItem={this.modifyItem}
                            onDeleteItem={this.deleteItem}
                        />
                    }
                    {
                        tabView === CHART_VIEW &&
                        <h1>This area for charts</h1>
                    }

                </div>
            </Fragment>

        )
    }
}

export default withRouter(withContext(Home));