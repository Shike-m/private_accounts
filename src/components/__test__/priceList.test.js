import React from 'react';
import { shallow } from 'enzyme';
import PriceList from '../priceList';
import { items, categories } from '../../pages/home';
import { Icon } from 'antd';

const itemsWithCategory = items.map(item => {
    item.category = categories[item.cid];
    return item;
});

const props = {
    items: itemsWithCategory,
    onModifyItem: jest.fn(),
    onDeleteItem: jest.fn(),
}

let wraper;
describe('test PriceList component', () => {
    beforeEach(() => {
        wraper = shallow(<PriceList {...props} />)
    });
    it('should render the component to match the snapshot', () => {
        expect(wraper).toMatchSnapshot();
    });
    it('should render correct price items length', () => {
        expect(wraper.find('.list-group-item').length).toEqual(itemsWithCategory.length);
    });
    it('should render correct icon and price for each item', () => {
        const iconList = wraper.find('.list-group-item').first().find(Icon)
        expect(iconList.length).toEqual(3);
        expect(iconList.first().props().type).toEqual(itemsWithCategory[0].category.iconName);
    });
    it('should triger the correct function callback', () => {
        const firstItem = wraper.find('.list-group-item').first();
        firstItem.find('a').first().simulate('click');
        expect(props.onModifyItem).toHaveBeenCalledWith(itemsWithCategory[0]);
        firstItem.find('a').last().simulate('click');
        expect(props.onDeleteItem).toHaveBeenCalledWith(itemsWithCategory[0]);
    })
})