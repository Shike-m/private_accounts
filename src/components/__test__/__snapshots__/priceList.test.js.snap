// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PriceList component should render the component to match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PriceList
    items={
      Array [
        Object {
          "category": Object {
            "iconName": "car",
            "id": 1,
            "name": "trip",
            "type": "outcome",
          },
          "cid": 1,
          "date": "2019-03-03",
          "id": 1,
          "price": 100,
          "title": "Travel to abroad",
        },
        Object {
          "category": Object {
            "iconName": "car",
            "id": 1,
            "name": "trip",
            "type": "outcome",
          },
          "cid": 1,
          "date": "2019-02-03",
          "id": 2,
          "price": 200,
          "title": "Travel to abroad",
        },
      ]
    }
    onDeleteItem={[MockFunction]}
    onModifyItem={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="list-group-item 
 d-flex justify-content-between align-items-center"
        >
          <span
            className="col-1"
          >
            <Icon
              className="rounded-circle"
              color="#000"
              fontSize="30px"
              style={
                Object {
                  "backgroundColor": "#007bff",
                  "padding": "5px",
                }
              }
              type="car"
            />
            trip
          </span>
          <span
            className="col-5"
          >
            Travel to abroad
          </span>
          <span
            className="col-2 font-weight-bold"
          >
            -
            100
          </span>
          <span
            className="col-2"
          >
            2019-03-03
          </span>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Icon
              className="rounded-circle"
              color="#000"
              fontSize="30px"
              style={
                Object {
                  "backgroundColor": "#28a745",
                  "padding": "5px",
                }
              }
              type="form"
            />
          </a>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Icon
              className="rounded-circle"
              color="#000"
              fontSize="30px"
              style={
                Object {
                  "backgroundColor": "#dc3545",
                  "padding": "5px",
                }
              }
              type="delete"
            />
          </a>
        </li>,
        <li
          className="list-group-item 
 d-flex justify-content-between align-items-center"
        >
          <span
            className="col-1"
          >
            <Icon
              className="rounded-circle"
              color="#000"
              fontSize="30px"
              style={
                Object {
                  "backgroundColor": "#007bff",
                  "padding": "5px",
                }
              }
              type="car"
            />
            trip
          </span>
          <span
            className="col-5"
          >
            Travel to abroad
          </span>
          <span
            className="col-2 font-weight-bold"
          >
            -
            200
          </span>
          <span
            className="col-2"
          >
            2019-02-03
          </span>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Icon
              className="rounded-circle"
              color="#000"
              fontSize="30px"
              style={
                Object {
                  "backgroundColor": "#28a745",
                  "padding": "5px",
                }
              }
              type="form"
            />
          </a>
          <a
            className="col-1"
            onClick={[Function]}
          >
            <Icon
              className="rounded-circle"
              color="#000"
              fontSize="30px"
              style={
                Object {
                  "backgroundColor": "#dc3545",
                  "padding": "5px",
                }
              }
              type="delete"
            />
          </a>
        </li>,
      ],
      "className": "list-group list-group-flush",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="col-1"
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  }
                }
                type="car"
              />
              trip
            </span>,
            <span
              className="col-5"
            >
              Travel to abroad
            </span>,
            <span
              className="col-2 font-weight-bold"
            >
              -
              100
            </span>,
            <span
              className="col-2"
            >
              2019-03-03
            </span>,
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                type="form"
              />
            </a>,
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                type="delete"
              />
            </a>,
          ],
          "className": "list-group-item 
 d-flex justify-content-between align-items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    }
                  }
                  type="car"
                />,
                "trip",
              ],
              "className": "col-1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "rounded-circle",
                  "color": "#000",
                  "fontSize": "30px",
                  "style": Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  },
                  "type": "car",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "trip",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Travel to abroad",
              "className": "col-5",
            },
            "ref": null,
            "rendered": "Travel to abroad",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "-",
                100,
              ],
              "className": "col-2 font-weight-bold",
            },
            "ref": null,
            "rendered": Array [
              "-",
              100,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2019-03-03",
              "className": "col-2",
            },
            "ref": null,
            "rendered": "2019-03-03",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                type="form"
              />,
              "className": "col-1",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "rounded-circle",
                "color": "#000",
                "fontSize": "30px",
                "style": Object {
                  "backgroundColor": "#28a745",
                  "padding": "5px",
                },
                "type": "form",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                type="delete"
              />,
              "className": "col-1",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "rounded-circle",
                "color": "#000",
                "fontSize": "30px",
                "style": Object {
                  "backgroundColor": "#dc3545",
                  "padding": "5px",
                },
                "type": "delete",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="col-1"
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  }
                }
                type="car"
              />
              trip
            </span>,
            <span
              className="col-5"
            >
              Travel to abroad
            </span>,
            <span
              className="col-2 font-weight-bold"
            >
              -
              200
            </span>,
            <span
              className="col-2"
            >
              2019-02-03
            </span>,
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                type="form"
              />
            </a>,
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                type="delete"
              />
            </a>,
          ],
          "className": "list-group-item 
 d-flex justify-content-between align-items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    }
                  }
                  type="car"
                />,
                "trip",
              ],
              "className": "col-1",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "rounded-circle",
                  "color": "#000",
                  "fontSize": "30px",
                  "style": Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  },
                  "type": "car",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "trip",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Travel to abroad",
              "className": "col-5",
            },
            "ref": null,
            "rendered": "Travel to abroad",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "-",
                200,
              ],
              "className": "col-2 font-weight-bold",
            },
            "ref": null,
            "rendered": Array [
              "-",
              200,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2019-02-03",
              "className": "col-2",
            },
            "ref": null,
            "rendered": "2019-02-03",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                type="form"
              />,
              "className": "col-1",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "rounded-circle",
                "color": "#000",
                "fontSize": "30px",
                "style": Object {
                  "backgroundColor": "#28a745",
                  "padding": "5px",
                },
                "type": "form",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                type="delete"
              />,
              "className": "col-1",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "rounded-circle",
                "color": "#000",
                "fontSize": "30px",
                "style": Object {
                  "backgroundColor": "#dc3545",
                  "padding": "5px",
                },
                "type": "delete",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="list-group-item 
 d-flex justify-content-between align-items-center"
          >
            <span
              className="col-1"
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  }
                }
                type="car"
              />
              trip
            </span>
            <span
              className="col-5"
            >
              Travel to abroad
            </span>
            <span
              className="col-2 font-weight-bold"
            >
              -
              100
            </span>
            <span
              className="col-2"
            >
              2019-03-03
            </span>
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                type="form"
              />
            </a>
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                type="delete"
              />
            </a>
          </li>,
          <li
            className="list-group-item 
 d-flex justify-content-between align-items-center"
          >
            <span
              className="col-1"
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#007bff",
                    "padding": "5px",
                  }
                }
                type="car"
              />
              trip
            </span>
            <span
              className="col-5"
            >
              Travel to abroad
            </span>
            <span
              className="col-2 font-weight-bold"
            >
              -
              200
            </span>
            <span
              className="col-2"
            >
              2019-02-03
            </span>
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  }
                }
                type="form"
              />
            </a>
            <a
              className="col-1"
              onClick={[Function]}
            >
              <Icon
                className="rounded-circle"
                color="#000"
                fontSize="30px"
                style={
                  Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  }
                }
                type="delete"
              />
            </a>
          </li>,
        ],
        "className": "list-group list-group-flush",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="col-1"
              >
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    }
                  }
                  type="car"
                />
                trip
              </span>,
              <span
                className="col-5"
              >
                Travel to abroad
              </span>,
              <span
                className="col-2 font-weight-bold"
              >
                -
                100
              </span>,
              <span
                className="col-2"
              >
                2019-03-03
              </span>,
              <a
                className="col-1"
                onClick={[Function]}
              >
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#28a745",
                      "padding": "5px",
                    }
                  }
                  type="form"
                />
              </a>,
              <a
                className="col-1"
                onClick={[Function]}
              >
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#dc3545",
                      "padding": "5px",
                    }
                  }
                  type="delete"
                />
              </a>,
            ],
            "className": "list-group-item 
 d-flex justify-content-between align-items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Icon
                    className="rounded-circle"
                    color="#000"
                    fontSize="30px"
                    style={
                      Object {
                        "backgroundColor": "#007bff",
                        "padding": "5px",
                      }
                    }
                    type="car"
                  />,
                  "trip",
                ],
                "className": "col-1",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "rounded-circle",
                    "color": "#000",
                    "fontSize": "30px",
                    "style": Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    },
                    "type": "car",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "trip",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Travel to abroad",
                "className": "col-5",
              },
              "ref": null,
              "rendered": "Travel to abroad",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "-",
                  100,
                ],
                "className": "col-2 font-weight-bold",
              },
              "ref": null,
              "rendered": Array [
                "-",
                100,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2019-03-03",
                "className": "col-2",
              },
              "ref": null,
              "rendered": "2019-03-03",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#28a745",
                      "padding": "5px",
                    }
                  }
                  type="form"
                />,
                "className": "col-1",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "rounded-circle",
                  "color": "#000",
                  "fontSize": "30px",
                  "style": Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  },
                  "type": "form",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#dc3545",
                      "padding": "5px",
                    }
                  }
                  type="delete"
                />,
                "className": "col-1",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "rounded-circle",
                  "color": "#000",
                  "fontSize": "30px",
                  "style": Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  },
                  "type": "delete",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="col-1"
              >
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    }
                  }
                  type="car"
                />
                trip
              </span>,
              <span
                className="col-5"
              >
                Travel to abroad
              </span>,
              <span
                className="col-2 font-weight-bold"
              >
                -
                200
              </span>,
              <span
                className="col-2"
              >
                2019-02-03
              </span>,
              <a
                className="col-1"
                onClick={[Function]}
              >
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#28a745",
                      "padding": "5px",
                    }
                  }
                  type="form"
                />
              </a>,
              <a
                className="col-1"
                onClick={[Function]}
              >
                <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#dc3545",
                      "padding": "5px",
                    }
                  }
                  type="delete"
                />
              </a>,
            ],
            "className": "list-group-item 
 d-flex justify-content-between align-items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Icon
                    className="rounded-circle"
                    color="#000"
                    fontSize="30px"
                    style={
                      Object {
                        "backgroundColor": "#007bff",
                        "padding": "5px",
                      }
                    }
                    type="car"
                  />,
                  "trip",
                ],
                "className": "col-1",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "rounded-circle",
                    "color": "#000",
                    "fontSize": "30px",
                    "style": Object {
                      "backgroundColor": "#007bff",
                      "padding": "5px",
                    },
                    "type": "car",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "trip",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Travel to abroad",
                "className": "col-5",
              },
              "ref": null,
              "rendered": "Travel to abroad",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "-",
                  200,
                ],
                "className": "col-2 font-weight-bold",
              },
              "ref": null,
              "rendered": Array [
                "-",
                200,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2019-02-03",
                "className": "col-2",
              },
              "ref": null,
              "rendered": "2019-02-03",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#28a745",
                      "padding": "5px",
                    }
                  }
                  type="form"
                />,
                "className": "col-1",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "rounded-circle",
                  "color": "#000",
                  "fontSize": "30px",
                  "style": Object {
                    "backgroundColor": "#28a745",
                    "padding": "5px",
                  },
                  "type": "form",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className="rounded-circle"
                  color="#000"
                  fontSize="30px"
                  style={
                    Object {
                      "backgroundColor": "#dc3545",
                      "padding": "5px",
                    }
                  }
                  type="delete"
                />,
                "className": "col-1",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "rounded-circle",
                  "color": "#000",
                  "fontSize": "30px",
                  "style": Object {
                    "backgroundColor": "#dc3545",
                    "padding": "5px",
                  },
                  "type": "delete",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
